tools/build/libwebrtc/build.ps1

```diff
@@ -0,0 +1,282 @@
+
+# Get the root folder of MixedReality-WebRTC
+function Get-RootFolder
+{
+    $repoRoot = Join-Path $PSScriptRoot "../../../" -Resolve
+    return $repoRoot
+}
+
+# Get the <root>/external/ folder
+function Get-ExternalFolder
+{
+    $repoRoot = Get-RootFolder
+    $externalFolder = Join-Path $repoRoot "external/"
+    return $externalFolder
+}
+
+# Write a message for a task starting, with a specific color to make it
+# stand out between long-running tasks themselves outputting lots of info.
+function Write-TaskStart([string]$message)
+{
+    Write-Host -Foreground Magenta "`n$message"
+}
+
+# Write a failure message without generating any Powershell error.
+function Write-Failure([string]$message)
+{
+    Write-Host -NoNewLine -Foreground Red "Error: "
+    Write-Host $message
+}
+
+# Write a key-value pair with higlight on the value.
+function Write-KeyValue([string]$key, [string]$value)
+{
+    Write-Host -NoNewLine $key
+    Write-Host -Foreground Blue $value
+}
+
+# Initialize the environment variables
+function Initialize-BuildEnvironment
+{
+    Write-TaskStart "Setting up environment..."
+
+    $repoRoot = Get-RootFolder
+    Write-KeyValue "Repository root: " $repoRoot
+
+    # Add depot_tools folder to PATH
+    $externalFolder = Join-Path $repoRoot "external/"
+    $depotToolsPath = Join-Path $externalFolder "depot_tools"
+    if (!($env:PATH.Contains($depotToolsPath)))
+    {
+        # Be sure to append, not prepend. depot_tools/ contains some git version,
+        # and prepending will make that version the default one, which may confuse
+        # some users as e.g. on Windows it won't have the credential manager set up
+        # like the one the user had installed already.
+        $env:PATH = "$env:PATH;$depotToolsPath"
+        Write-Host "Updated PATH to include depot_tools location"
+    }
+    else
+    {
+        Write-Host "PATH already includes depot_tools location"
+    }
+
+    Write-Host  "Use local Windows toolchain (DEPOT_TOOLS_WIN_TOOLCHAIN = 0)"
+    $env:DEPOT_TOOLS_WIN_TOOLCHAIN = "0"
+
+    Write-Host "Use Visual Studio 2019 (GYP_MSVS_VERSION = 2019)"
+    $env:GYP_MSVS_VERSION = "2019"
+}
+
+# Install the Google repository of libwebrtc into external/libwebrtc
+function Install-GoogleRepository
+{
+    $externalFolder = Get-ExternalFolder
+
+    # Prepare the folder
+    $repoFolder = Join-Path $externalFolder "libwebrtc"
+    New-Item $repoFolder -ItemType Directory -Force | Out-Null
+    $repoFolder = Resolve-Path $repoFolder
+    Write-Host "Installing Google's libwebrtc in $repoFolder"
+
+    Push-Location $repoFolder
+
+    # Download
+    Write-TaskStart "Fetching Google repository, this might take some time..."
+    try {
+        Invoke-Expression "fetch --nohooks --no-history webrtc"
+    }
+    catch {
+        Write-Failure "Failed to fetch Google repository 'webrtc'."
+        throw
+    }
+    
+    # Checkout + sync
```

Comment:
We should separate checkout script from build script so one can build with local changes to WebRTC. Can be done as a separate change though.

Reply:
Agreed.